ash/src/extensions/ext/debug_utils.rs

```diff
@@ -7,22 +7,16 @@ use std::mem;
 #[derive(Clone)]
 pub struct DebugUtils {
     handle: vk::Instance,
-    debug_utils_fn: vk::ExtDebugUtilsFn,
+    fns: vk::ExtDebugUtilsFn,
 }
 
 impl DebugUtils {
     pub fn new(entry: &Entry, instance: &Instance) -> Self {
-        let debug_utils_fn = vk::ExtDebugUtilsFn::load(|name| unsafe {
-            mem::transmute(entry.get_instance_proc_addr(instance.handle(), name.as_ptr()))
+        let handle = instance.handle();
+        let fns = vk::ExtDebugUtilsFn::load(|name| unsafe {
+            mem::transmute(entry.get_instance_proc_addr(handle, name.as_ptr()))
```

Comment:
Almost all the calls here take a `device` or a child of the `device` (`CommandBuffer`, `Queue`) so we might as well change this to `get_device_proc_addr` and hold on to `device` in addition to `instance`. However, the `_messenger` and `_message` calls are _not_ device specific and AFAIK it is not allowed to query these through `get_device_proc_addr`. We could think something up for these (in a future PR) or accept the fact that these might wrap a jump table and that the device needs to be passed to those specific functions, as we cannot easily rework the `load` function.

A future PR may also try to split the `load` up in functions that receive a `device` or child of the `device`, so that we can load them separately.